### DISCUSSION TITLE
How to call Hallucination Guardrail

### DISCUSSION THREAD

**Discussion by PeerBoerner1** [NVIDIA-NeMo/Guardrails#609] [General]
I am having difficulty figuring out how to call the "self check hallucinations" guardrail. The documentation says this is an Output Guardrail (makes sense) but the example shows it as in input rail.  Also, when I set up self check hallucinations it says there is no output rail flow with that name. The same is true if I use "self check hallucination" (singular).  When I check the hallucination flow file provided (nemoguardrails/library/hallucination/flows.co) the only flow I see that would be usable as an output rail name would be "hallucination warning" (define flow hallucination warning) which does get loaded without error. BUT when I run my code, I see the Hallucination warning rail start but never complete.  Has anyone had any luck using this rail? I'd appreciate any guidance you can provide. 

Also, is the flows.co file in the nemoguardrails/library/hallucination folder an example or are those the flows I must reference in my flow definitions?

**Comment by sjay8** [comment]
@PeerBoerner1 Hi peer! did you figure out this issue? having similar issues like you
